plugins/net_plugin/net_plugin.cpp

```diff
@@ -2905,6 +2929,11 @@ namespace eosio {
       // use c in this method instead of this to highlight that all methods called on c-> must be thread safe
       connection_ptr c = shared_from_this();
 
+      {
+         std::lock_guard<std::mutex> g_conn( conn_mtx );
+         c->last_block = blk_num;
+      }
```

Comment:
Since this block might be rejected. Move the recording of `last_accepted_block`  down below so it reports only validated accepted blocks.